exercises/exercise_4/README.md

```diff
@@ -1,248 +1,190 @@
 
-* **Display** the status of each installation
 
-```sh
-$ docker-app status my-hello-installation
-ID                  NAME                          MODE                REPLICAS            IMAGE                        PORTS
-qspyvbdxf1js        my-hello-installation_hello   replicated          1/1                 hashicorp/http-echo:latest   *:8080->8080/tcp
+## Docker App Parameters
 
-$ docker-app status my-hello-prod-installation
-ID                  NAME                               MODE                REPLICAS            IMAGE                        PORTS
-5d1glvey8b1l        my-hello-prod-installation_hello   replicated          1/1                 hashicorp/http-echo:latest   *:80->80/tcp
-```
+In addition to using environment variables, Docker App allows you to define parameters for an application that can be overridden at deploy-time. Examples of parameters might include port mappings, label definitions, or network names. _Almost_ everything in the Docker App compose file can be parameterized, except for the image names.
 
-### Upgrade
 
-`Usage:  docker-app upgrade <installation-name> [options] [flags]`
+## Using a Parameter to Change Voting Options
 
-`upgrade` command lets you modify your installation, for example to change a port or to scale a service.
+By default, the vote and results services let you vote between Dogs and Cats. However, wouldn't it be neat if we could change that? Conveniently, both the `vote` and `results` services allow the definition of two environment variables (`OPTION_A` and `OPTION_B`) to let us change the voting choices. This sounds like a great opportunity for parameters!
 
-* **`upgrade`** the `my-hello-installation` installation and change the `hello.text`
+1. In both the `vote` and `results` services, add environment variables named `OPTION_A` and `OPTION_B` with a value of `${optionA}` and `${optionB}`. The values are simply placeholders for the parameters.
 
-```sh
-$ docker-app upgrade my-hello-installation -s hello.text="Hello upgrade"
-Updating service my-hello-installation_hello (id: qspyvbdxf1jsh5hj95er7x598)
-```
-
-* **Check** the text has been updated on port `8080`
-
-### Uninstall
-
-`Usage:  docker-app uninstall <installation-name> [flags]`
-
-`uninstall` command removes your installation.
-
-* **`uninstall`** all the previous installations.
-
-```sh
-$ docker-app uninstall my-hello-installation
-Removing service my-hello-installation_hello
-Removing network my-hello-installation_default
-```
-
-## Summary
-
-* The bundle is a new way to install your application. It is self-contained, including the application files and the deploy runtime binaries.
-* The invocation image can be stored and shared in a registry
-* The bundle.json is a portable way to define the application
+    <details>
+      <summary>Solution</summary>
+    
+    ```yaml
+    services:
+      vote:
+        environment:
+          OPTION_A: ${optionA}
```

Comment:
And what about using the hierarchical key system, with something like `${option.A}` and `${option.B}` ?
And then in the `parameters.yml` file:
```yaml
option:
  A: Cats
  B: Dogs
```